### PR TITLE
fix(many): border colors are adjusted to meet a11y contrast standards

### PR DESCRIPTION
**Pull request by ToMESSKa**
Closes: INSTUI-4319

TEST PLAN:
- please check if all the necessary adjustments were made (except Drilldown/Menu/MenuItem selected row background color and ColorPicker IconButton border color - these will be moved to a new pull request): https://www.figma.com/design/8Ai2uBfoktyApNWHMBZFMu/%E2%9C%85-%5BDelivered%5D-Required%2C-error%2C-A11y-handover?node-id=1-2094&node-type=canvas&t=NuZAJJEIZDK8MBgS-0
- check if colors are changed in 'canvas-high-contrast' theme in the docs

### CODE REVIEW

File: packages/ui-checkbox/src/Checkbox/ToggleFacade/styles.ts [hunk 1]
```diff
@@ -158,13 +158,16 @@ const generateStyle = (
...
-        left: componentTheme.borderWidth,
-        height: `calc(100% - (${componentTheme.borderWidth} * 2))`,
-        width: `calc(100% - (${componentTheme.borderWidth} * 2))`,
+        height: `calc(100% - (${componentTheme.borderWidth} * 6))`,
+        width: `calc(100% - (${componentTheme.borderWidth} * 6))`,
         background: componentTheme.toggleBackground,
         boxShadow: componentTheme.toggleShadow,
-        borderRadius: '100%'
+        borderRadius: '100%',
+        border: componentTheme.borderWidth + ' solid',
```
Comment (balzss) [inline packages/ui-checkbox/src/Checkbox/ToggleFacade/styles.ts:166]: you can write this a bit simpler:
```
const { borderWidth, uncheckedIconBorderColor, checkedIconBorderColor} = componentTheme
...
border: `${borderWidth} solid ${checked ? 'checkedIconBorderColor' : 'uncheckedIconBorderColor'}`,
```

File: packages/ui-drilldown/src/Drilldown/index.tsx [hunk 2]
```diff
@@ -1377,6 +1377,10 @@ class Drilldown extends Component<DrilldownProps, DrilldownState> {
           getDisabledOptionProps
         }) => (
           <View
+            themeOverride={{
```
Comment (balzss) [inline packages/ui-drilldown/src/Drilldown/index.tsx:1380]: we shouldn't use `themeOverride` for our own components

File: packages/ui-drilldown/src/Drilldown/index.tsx [hunk 3]
```diff
@@ -1418,6 +1422,9 @@ class Drilldown extends Component<DrilldownProps, DrilldownState> {
             })}
           >
             <View
+              themeOverride={{
```
Comment (balzss) [inline packages/ui-drilldown/src/Drilldown/index.tsx:1425]: we shouldn't use `themeOverride` for our own components

File: packages/ui-popover/src/Popover/index.tsx [hunk 4]
```diff
@@ -570,7 +576,17 @@ class Popover extends Component<PopoverProps, PopoverState> {
               : placement
         } as Partial<ContextViewProps> & { ref: any }
 
-        return <ContextView {...viewProps}>{content}</ContextView>
+        return (
+          <ContextView
+            {...viewProps}
+            themeOverride={{
```
Comment (balzss) [inline packages/ui-popover/src/Popover/index.tsx:582]: we shouldn't use `themeOverride` for our own components

File: packages/ui-popover/src/Popover/index.tsx [hunk 5]
```diff
@@ -579,7 +595,14 @@ class Popover extends Component<PopoverProps, PopoverState> {
           ...(color === 'primary-inverse' && { borderColor: 'transparent' })
         } as Partial<ViewProps> & { ref: any }
 
-        return <View {...viewProps}>{content}</View>
+        return (
+          <View
+            {...viewProps}
+            themeOverride={{ borderColorPrimary: styles?.borderColor }}
```
Comment (balzss) [inline packages/ui-popover/src/Popover/index.tsx:601]: we shouldn't use `themeOverride` for our own components

File: packages/ui-toggle-details/src/ToggleGroup/index.tsx [hunk 6]
```diff
@@ -162,6 +174,7 @@ class ToggleGroup extends Component<ToggleGroupProps> {
               display="block"
               borderRadius="medium"
               background="primary"
+              themeOverride={{ borderColorPrimary: styles?.borderColor }}
```
Comment (balzss) [inline packages/ui-toggle-details/src/ToggleGroup/index.tsx:177]: we shouldn't use `themeOverride` for our own components

**Review by balzss — CHANGES_REQUESTED** [review]

File: packages/ui-checkbox/src/Checkbox/ToggleFacade/styles.ts (hunk 1 above)
Comment (ToMESSKa) [reply packages/ui-checkbox/src/Checkbox/ToggleFacade/styles.ts]: rewrote that part

File: packages/ui-drilldown/src/Drilldown/index.tsx (hunk 3 above)
Comment (ToMESSKa) [reply packages/ui-drilldown/src/Drilldown/index.tsx]: I think according to our last discussion with Topi, we haven't made a conclusion that themeOverride should be used or not

**Review by balzss — APPROVED** [review]

File: packages/ui-toggle-details/src/ToggleGroup/index.tsx (hunk 6 above)
Comment (ToMESSKa) [reply packages/ui-toggle-details/src/ToggleGroup/index.tsx]: removed

File: packages/ui-popover/src/Popover/index.tsx (hunk 5 above)
Comment (ToMESSKa) [reply packages/ui-popover/src/Popover/index.tsx]: removed

File: packages/ui-popover/src/Popover/index.tsx (hunk 4 above)
Comment (ToMESSKa) [reply packages/ui-popover/src/Popover/index.tsx]: removed

File: packages/ui-drilldown/src/Drilldown/index.tsx (hunk 3 above)
Comment (ToMESSKa) [reply packages/ui-drilldown/src/Drilldown/index.tsx]: removed

File: packages/ui-drilldown/src/Drilldown/index.tsx (hunk 2 above)
Comment (ToMESSKa) [reply packages/ui-drilldown/src/Drilldown/index.tsx]: removed

**Review by balzss — APPROVED** [review]
looks good, nice work!

**Review by HerrTopi — APPROVED** [review]

**Pull request merged**